lib/validate/validate_function.js

```diff
@@ -76,10 +77,11 @@ module.exports = function validateFunction(options) {
                 objectElementValidators: { zoom: validateNumber, value: validateValue }
             }));
         } else {
-            errors = errors.concat(validate({
+            var isZoomFunction = originalValue.property === '$zoom' || !originalValue.property;
```

Comment:
I think this should be just `var isZoomFunction = !originalValue.property;`

`$zoom` isn't supported by mapbox-gl-function